pkg/webui/containers/collaborator-select/index.js

```diff
@@ -79,47 +92,70 @@ const Suggest = ({ initialOptions, userInputCustomComponent, entity, entityId, .
   const handleLoadingOptions = useCallback(
     async value => {
       if (Boolean(value)) {
-        await dispatch(attachPromise(searchAccounts(value, onlyUsers, collaboratorOf)))
-        const newOptions = searchResultsRef.current.map(account => ({
-          value:
-            'user_ids' in account
-              ? account.user_ids?.user_id
-              : account.organization_ids?.organization_id,
-          label:
-            'user_ids' in account
-              ? account.user_ids?.user_id
-              : account.organization_ids?.organization_id,
-          icon: 'user_ids' in account ? 'user' : 'organization',
-        }))
-        const translatedOptions = newOptions?.map(option => {
-          const { label, labelValues = {} } = option
-          if (typeof label === 'object' && label.id && label.defaultMessage) {
-            return { ...option, label: formatMessage(label, labelValues) }
-          }
-
-          return option
-        })
-
-        return translatedOptions
+        try {
+          await dispatch(attachPromise(searchAccounts(value, onlyUsers, collaboratorOf)))
+          const newOptions = searchResultsRef?.current?.map(account => ({
+            value:
+              'user_ids' in account
+                ? account.user_ids?.user_id
+                : account.organization_ids?.organization_id,
+            label:
+              'user_ids' in account
+                ? account.user_ids?.user_id
+                : account.organization_ids?.organization_id,
```

Comment:
@ryaplots I'm did the same thing for a similar situation here:
https://github.com/TheThingsNetwork/lorawan-stack/blob/d068363dc0def9b48b95d17e2cf11d37c069b9b2/pkg/webui/console/store/middleware/logics/packet-broker.js#L43-L61

Reply:
Isn't it easier to just return the name as well in the original search request? @adriansmares @kschiffer

Reply:
Organizations can be contacts for other entities, so it is not a mistake to also search for organizations when it comes to contacts. Think of the following use case - you have a fleet of gateways and you would like to set the technical contact for all of them. If you set a user, you will have to update all of them later on, but if you set an organization, you just need to update the technical contact in that organization.

---

Regarding the `name`: the problem is that these are two separate types of entities: accounts and users - the fact that they have some shared attributes (`name`, `description`, `attributes`) does not change that the entity types can be different - it is not as straight forward as it sounds at face value.

Now this doesn't mean that it is not a solvable problem - we can consider having in the API an entity which contains these common attributes, and then adding a field mask to the search request which would be able to return only these common fields. With that being said, I would prefer not to stall this functionality now and rather implement this in a subsequent PR. I do want to have this in the API however - I agree that it would be useful.

Reply:
Alright. Then let's solve it by an additional API request like I suggested for now and create an issue for extending the account search API. @ryaplots can you do that?

You can just add this to the account search store logic. After fetching the accounts, collect the IDs, then cycle through the collaborator list and decorate the account search result and filter by the IDs to collect the collaborator names. Ideally, the collaborator names would be cached somehow so that we don't need to launch the request all the time.